selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/preferences/ContributeTabTest.java

```diff
@@ -0,0 +1,288 @@
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private NotificationsPopupPanel notificationsPopupPanel;
+  @Inject private PanelSelector panelSelector;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    Path entryPath =
+        Paths.get(getClass().getResource("/projects/default-spring-project").getPath());
+    testRepo.addContent(entryPath);
+    testRepo2.addContent(entryPath);
+
+    URL resource = getClass().getResource("/projects/guess-project");
+    testProjectServiceClient.importProject(
+        testWorkspace.getId(),
+        Paths.get(resource.toURI()),
+        THIRD_PROJECT_NAME,
+        ProjectTemplates.MAVEN_SPRING);
+
+    ide.open(testWorkspace);
+  }
+
+  @AfterMethod
+  public void closeForm() {
+    try {
+      if (askDialog.isOpened()) {
+        askDialog.confirmAndWaitClosed();
+      }
+
+      if (preferences.isPreferencesFormOpened()) {
+        preferences.clickOnCloseButton();
+      }
+
+      preferences.waitPreferencesFormIsClosed();
+
+    } catch (Exception e) {
+      LOG.error(e.getLocalizedMessage(), e);
+    }
+  }
+
+  @Test
+  public void checkSettingContributeTab() {
+    firstProjectUrl = testRepo.getHtmlUrl() + ".git";
+    secondProjectUrl = testRepo2.getHtmlUrl() + ".git";
+
+    projectExplorer.waitProjectExplorer();
+    projectExplorer.waitItem(THIRD_PROJECT_NAME);
+
+    // import the first and second projects
+    git.importJavaApp(firstProjectUrl, FIRST_PROJECT_NAME, MAVEN);
+    git.importJavaApp(secondProjectUrl, SECOND_PROJECT_NAME, MAVEN);
+
+    // check opening the PR panel by select a project
+    pullRequestPanel.waitClosePanel();
+    projectExplorer.waitAndSelectItem(FIRST_PROJECT_NAME);
+    pullRequestPanel.waitOpenPanel();
+
+    // check the text when a project is not under VCS
+    projectExplorer.waitAndSelectItem(THIRD_PROJECT_NAME);
+    pullRequestPanel.waitTetxNotVcsProject(EXP_TEXT_NOT_VCS);
+
+    // check the 'Contribute' checkbox is true by default
+    openContributeTab();
+    preferences.waitContributeCheckboxIsSelected();
+    preferences.clickOnCloseBtnAndWaitClosedForm();
+  }
+
+  @Test(priority = 1)
+  public void checkRefreshAndSaveButton() {
+    openContributeTab();
+
+    preferences.setAndWaitStateContributeCheckbox(true);
+    preferences.clickOnOkBtn();
+
+    assertFalse(preferences.saveButtonIsEnabled());
+
+    // check the 'Refresh button
+    preferences.clickOnContributeCheckbox();
+    preferences.waitContributeCheckboxIsNotSelected();
+
+    assertTrue(preferences.saveButtonIsEnabled());
+
+    preferences.clickRefreshButton();
+    preferences.waitContributeCheckboxIsSelected();
+
+    assertFalse(
+        preferences.saveButtonIsEnabled(),
+        "Known issue https://github.com/eclipse/che/issues/9959");
+
+    preferences.clickOnCloseBtnAndWaitClosedForm();
+  }
+
+  @Test(priority = 1)
+  public void checkSwitchProjectsWhenContributeIsFalse() {
```

Comment:
What "Contribute is false" mean?

Reply:
It means that the checkbox of the 'Contribute' is turn off and the property "Activate Contribute Panel by clicking on a project" is false. See the PR: https://github.com/eclipse/che/pull/9806